modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/planner/IndexRebuildPlannerTest.java

```diff
@@ -92,6 +95,39 @@ public void testIndexCountAtIndexRebuild() throws Exception {
             .and(nodeOrAnyChild(isInstanceOf(IgniteIndexCount.class)))));
     }
 
+    /** Test Index min/max (first/last) is disabled when index is unavailable. */
+    @Test
+    public void testIndexMinMaxAtIndexRebuild() throws Exception {
+        checkMinMaxOptimized();
+
+        tbl.markIndexRebuildInProgress(true);
+
+        assertPlan("SELECT MIN(VAL) FROM TBL", publicSchema, isInstanceOf(IgniteAggregate.class)
+            .and(a -> a.getAggCallList().stream().filter(agg -> agg.getAggregation().getKind() == MIN).count() == 1)
+            .and(nodeOrAnyChild(isInstanceOf(IgniteTableScan.class))));
```

Comment:
It's better to use `hasChildThat` than `nodeOrAnyChild` here (current node can't be `IgniteTableScan`, since it is `IgniteAggregate`)
THe same for `checkMinMaxOptimized` method.
`isInstanceOf(IgniteTableScan.class)` can also be optimized to `isTableScan(tbl)`